posts/2024-02-13-24.0.0.2-beta.adoc

```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+title: "Back-channel logout support for OpenID Connect clients and servers in 24.0.0.2-beta"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/dmuelle
+author_github: https://github.com/dmuelle
+seo-title: Back-channel logout support for OpenID Connect clients and servers in 24.0.0.2-beta - OpenLiberty.io
+seo-description: OpenID Connect clients and servers in Open Liberty now support back-channel logout. Back-channel logout allows OpenID Connect servers to directly notify OpenID Connect clients of a user logout so each OpenID Connect client can also log the user out locally.
+blog_description: OpenID Connect clients and servers in Open Liberty now support back-channel logout. Back-channel logout allows OpenID Connect servers to directly notify OpenID Connect clients of a user logout so each OpenID Connect client can also log the user out locally.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Back-channel logout support for OpenID Connect clients and servers in 24.0.0.2-beta
+David Mueller <https://github.com/dmuelle>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post
+
+OpenID Connect clients and servers in Open Liberty now support back-channel logout. Back-channel logout allows OpenID Connect servers to directly notify OpenID Connect clients of a user logout so each OpenID Connect client can also log the user out locally.
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/27362
+// Contact/Reviewer: jimmy1wu
+// // // // // // // //
+
+Previously, OpenID Connect servers could notify OpenID Connect clients that a user logged out only by using iframes that were embedded in the OpenID Connect client's web page. If the web page wasn't active, the OpenID Connect client wasn't notified of the logout that occurred on the OpenID Connect server. Back-channel logout solves this problem through direct communication between the OpenID Connect server and clients.
+
+== Back-channel logout for OpenID Connect clients
+
+With this release, back-channel logout for OpenID clients is enabled by default when the link:{url-prefix}/docs/latest/reference/feature/openidConnectClient.html[OpenID Connect Client] and link:{url-prefix}/docs/latest/reference/feature/socialLogin.html[Social Media Login] features are enabled. The following back-channel logout endpoints are enabled for those features:
```

Comment:
can we provide a little more context for these endpoints? what does the dev do with them? They're automatically enabled on the client- but is it the server that actually needs to code them? It's not clear if you dont already know what theyre for

Reply:
how about:

```
With this release, back-channel logout for OpenID Connect clients is enabled by default when the link:{url-prefix}/docs/latest/reference/feature/openidConnectClient.html[OpenID Connect Client] and link:{url-prefix}/docs/latest/reference/feature/socialLogin.html[Social Media Login] features are enabled. This means that the back-channel logout endpoints are automatically enabled when an `openidConnectClient` or a `socialLogin` element is defined in the `server.xml` file. The back-channel logout endpoint is the endpoint on the OpenID Connect client which the OpenID Connect server will use to send back-channel logout requests to in order to log out the user at the OpenID Connect client. The OpenID Connect server will need to know the absolute URI of the back-channel logout endpoint in order to send the back-channel logout requests. The following back-channel logout endpoints are enabled for those features:

* OpenID Connect Client: `/oidcclient/backchannel_logout/{oidcClientId}`
* Social Media Login: `/ibm/api/social-login/backchannel_logout/{socialLoginId}`

For example when using OpenID Connect Client, the back-channel logout endpoint will be `/oidcclient/backchannel_logout/client01` for the following configuration:

[source,xml]
----
<featureManager>
    <feature>openidConnectClient-1.0</feature>
</featureManager>
...
<openidConnectClient id="client01" ... />
----

For example when using Social Media Login, the back-channel logout endpoint will be `/ibm/api/social-login/backchannel_logout/client01` for the following configuration:

[source,xml]
----
<featureManager>
    <feature>socialLogin-1.0</feature>
</featureManager>
...
<oidcLogin id="client01" ... />
----
```

Reply:
updated